<?xml version="1.0"?>
<launch>

  <!-- Gazebo & GUI Configuration -->
  <arg name="headless"    default="false" doc="Should the gazebo GUI be launched?" />
  <arg name="paused"      default="false" doc="Should the simulation directly be stopped at 0s?" />
  <arg name="rviz"        default="false" doc="Should RVIz be launched?" />

  <!-- Robot Customization -->
  <arg name="arm_id"      default="panda" doc="Name of the panda robot to spawn" />
  <arg name="use_gripper" default="true"  doc="Should a franka hand be mounted on the flange?" />
  <arg name="controller"  default=" "     doc="Which controller should be started? (One of {cartesian_impedance,model,force,joint_position,joint_velocity}_controller)" />
  <arg name="x"           default="0"     doc="How far forward to place the base of the robot in [m]?" />
  <arg name="y"           default="0"     doc="How far leftwards to place the base of the robot in [m]?" />
  <arg name="z"           default="0"     doc="How far upwards to place the base of the robot in [m]?" />
  <arg name="roll"        default="0"     doc="How much to rotate the base of the robot around its X-axis in [rad]?" />
  <arg name="pitch"       default="0"     doc="How much to rotate the base of the robot around its Y-axis in [rad]?" />
  <arg name="yaw"         default="0"     doc="How much to rotate the base of the robot around its Z-axis in [rad]?" />

  <!-- load ijs controller names -->
  <rosparam command="load" file="$(find ijs_controllers)/config/ijs_controllers.yaml" subst_value="true" />

  <!-- controller specific sim workarounds -->
  <param name="/cartesian_impedance_controller/send_tau_immediately" value="true" />

  <!-- load franka_gazebo -->
  <include file="$(find franka_gazebo)/launch/panda.launch">
    <arg name="headless"    value="$(arg headless)" />
    <arg name="paused"      value="$(arg paused)" />
    <arg name="rviz"        value="$(arg rviz)" />
    <arg name="arm_id"      value="$(arg arm_id)" />
    <arg name="use_gripper" value="$(arg use_gripper)" />
    <arg name="controller"  value="$(arg controller)" />
    <arg name="x"           value="$(arg x)" />
    <arg name="y"           value="$(arg y)" />
    <arg name="z"           value="$(arg z)" />
    <arg name="roll"        value="$(arg roll)" />
    <arg name="pitch"       value="$(arg pitch)" />
    <arg name="yaw"         value="$(arg yaw)" />
  </include>

  <!-- RBS integration -->
  <include file="$(find ijs_controllers)/launch/blockset_integration.launch" >
      <arg name="arm_id" value="$(arg arm_id)" />
  </include>

</launch>